auth/authenticators/basic-authenticator.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2019 IBM Corp. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import extend = require('extend');
+import { computeBasicAuthHeader, validateInput } from '../utils';
+import { Authenticator } from './authenticator';
+import { AuthenticateOptions, AuthenticatorInterface } from './authenticator-interface';
+
+export type Options = {
+  username?: string;
+  password?: string;
```

Comment:
Why are username and password optional here?

Reply:
Wow good catch - glad I asked for a review! That's a mistake and I will fix right now

Reply:
fixed